code/src/main/kotlin/builtins/HclBuiltinFunctions.kt

```diff
@@ -199,6 +218,144 @@ private fun buildThenFunction() = buildFunction(
         returnType = Type.Bool,
         body = "if (condition) { body(); }\nreturn condition;"
 )
+
+private fun buildEachFunction() = buildFunction(
+        identifier = "each",
+        parameters = listOf(
+                Parameter("list", Type.List(Type.GenericType("T"))),
+                Parameter("body", Type.Func.ExplicitFunc(listOf(Type.GenericType("T")), Type.None))
+        ),
+        returnType = Type.None,
+        body = "for (int i = 0; i < list.get()->size; i++) {\n" +
+                "body(list.get()->data[i]);\n" +
+                "}\n" +
+                "return;"
+)
+
+private fun buildMapFunction() = buildFunction(
+        identifier = "map",
+        parameters = listOf(
+                Parameter("list", Type.List(Type.GenericType("T"))),
+                Parameter("fun", Type.Func.ExplicitFunc(
+                        listOf(Type.GenericType("T")),
+                        Type.GenericType("T"))
+                )
+        ),
+        returnType = Type.List(Type.GenericType("T")),
+        body = "T result[list.get()->size];\n" +
+                "for (int i = 0; i < list.get()->size; i++) {\n" +
+                "result[i] = fun(list.get()->data[i]);\n" +
+                "}\n" +
+                "return ConstList<T>::create(result, list.get()->size);"
+)
+
+private fun buildFilterFunction() = buildFunction(
+        identifier = "filter",
+        parameters = listOf(
+                Parameter("list", Type.List(Type.GenericType("T"))),
+                Parameter("fun",
+                        Type.Func.ExplicitFunc(
+                                listOf(Type.GenericType("T")),
+                                Type.Bool
+                        )
+                )
+        ),
+        returnType = Type.List(Type.GenericType("T")),
+        body = "T result[list.get()->size];\n" +
+                "int index = 0;\n" +
+                "for (int i = 0; i < list.get()->size; i++) {\n" +
+                "if (fun(list.get()->data[i]))\n" +
+                "result[index++] = list.get()->data[i];\n" +
+                "}\n" +
+                "return ConstList<T>::create(result, index);"
+)
+
+private fun buildDelayMillisFunction() = buildFunction(
+        identifier = "delayMillis",
+        parameters = listOf(Parameter("millis", Type.Number)),
+        returnType = Type.None,
+        body = "delayMillis(millis);\n" +
+                "return;"
+)
+
+//region PinFunctions
+private fun buildWriteDigPinHighFunction() = buildFunction(
```

Comment:
only make one method with two parameters, pin and state. (num, bool) called "setPin"